src/container/synology_container.py

```
@@ -291,9 +326,154 @@ def delete_image(self, name: str, tag: str = "latest") -> Dict[str, Any]:
+            return {"success": False, "error": {"code": "invalid_host", "message": "NAS URL has no hostname"}}
+
+        askpass_fd, askpass = tempfile.mkstemp(prefix="synology-mcp-askpass-")
+        os.close(askpass_fd)
+        try:
+            os.chmod(askpass, stat.S_IRWXU)
+            with open(askpass, "w", encoding="utf-8") as handle:
+                handle.write("#!/bin/sh\nprintf '%s\\n' \\\"$SYNOLOGY_SSH_PASSWORD\\\"\\n")
+            env = os.environ.copy()
+            env.update({"SYNOLOGY_SSH_PASSWORD": self._ssh_password, "SSH_ASKPASS": askpass, "SSH_ASKPASS_REQUIRE": "force", "DISPLAY": "synology-mcp"})
+            command = "sudo -n /var/packages/ContainerManager/target/usr/bin/docker " + docker_command
+            result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)